RadarSDK/Radar.m

```diff
@@ -303,28 +303,78 @@ + (RadarTripOptions *)getTripOptions {
 }
 
 + (void)startTripWithOptions:(RadarTripOptions *)options {
-    [RadarSettings setTripOptions:options];
-    [[RadarAPIClient sharedInstance] updateTripWithOptions:options status:RadarTripStatusStarted];
-    [[RadarLocationManager sharedInstance] getLocationWithCompletionHandler:nil];
+    [self startTripWithOptions:options completionHandler:nil];
 }
 
-+ (void)updateTripWithOptions:(RadarTripOptions *)options {
-    [[RadarAPIClient sharedInstance] updateTripWithOptions:options status:RadarTripStatusUnknown];
++ (void)startTripWithOptions:(RadarTripOptions *)options completionHandler:(RadarTripCompletionHandler)completionHandler {
+    [[RadarAPIClient sharedInstance] updateTripWithOptions:options
+                                                    status:RadarTripStatusStarted
+                                         completionHandler:^(RadarStatus status) {
+                                             if (status == RadarStatusSuccess) {
+                                                 [RadarSettings setTripOptions:options];
+
+                                                 [[RadarLocationManager sharedInstance] getLocationWithCompletionHandler:nil];
```

Comment:
same as the android PR, can we add a comment here explaining why getting a location is needed here, but not in the other `updateTrip` cases

Reply:
Thinking about it more, it's actually probably not a bad idea to flush a location update to the server in all cases. Will add and test.

Reply:
Done